src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/Addressable.java

```diff
@@ -30,33 +30,18 @@
  * a {@linkplain #address() memory address}. Examples of addressable types are {@link MemorySegment},
  * {@link MemoryAddress}, {@link VaList} and {@link CLinker.UpcallStub}.
  * <p>
- * An addressable instance is associated with a {@linkplain ResourceScope resource scope}; the resource scope determines the
- * lifecycle of the addressable instance, as well as whether the instance can be used from multiple threads. Some
- * addressable instances might not be associated with a lifecycle, in which case {@link #scope()} returns the
- * {@linkplain ResourceScope#globalScope() global scope}. Attempting to obtain a memory address
- * from an addressable instance whose backing scope has already been {@linkplain ResourceScope#isAlive() closed} always
- * results in an exception.
- * <p>
  * The {@link Addressable} type is used by the {@link CLinker C linker} to model the types of
  * {@link CLinker#downcallHandle(FunctionDescriptor) downcall handle} parameters that must be passed <em>by reference</em>
  * (e.g. memory addresses, va lists and upcall stubs).
  *
  * @implSpec
  * Implementations of this interface are <a href="{@docRoot}/java.base/java/lang/doc-files/ValueBased.html">value-based</a>.
  */
-public interface Addressable {
+public sealed interface Addressable permits MemorySegment, MemoryAddress, CLinker.UpcallStub, VaList {
 
     /**
      * Returns the memory address associated with this addressable.
      * @return The memory address associated with this addressable.
-     * @throws IllegalStateException if the scope associated with this addressable has been closed, or if access occurs from
-     * a thread other than the thread owning that scope.
```

Comment:
Assuming that this method can still throw an exception for the reason described here, is it okay for API methods to throw unspecified exceptions?

Reply:
for runtime exception, I think that's fine. That's kind of the case with `UnsupportedOperationException`. We might add some text to the documentation, but in general I think that's within bounds.